base/save.go

```diff
@@ -240,3 +246,43 @@ func GenerateAvailableName(r repo.Repo, peername, prefix string) string {
 		}
 	}
 }
+
+// Drop sets named components to nil
+func Drop(ds *dataset.Dataset, revStr string) error {
+	if revStr == "" {
+		return nil
+	}
+
+	revs, err := dsref.ParseRevs(revStr)
+	if err != nil {
+		return err
+	}
+
+	for _, rev := range revs {
+		if rev.Gen != 1 {
+			return fmt.Errorf("cannot drop specific generations")
+		}
+		switch rev.Field {
+		case "md":
+			ds.Meta = nil
+		case "vz":
+			ds.Viz = nil
+		case "tf":
+			ds.Transform = nil
+		case "st":
+			ds.Structure = nil
+		case "bd":
+			ds.Body = nil
+		case "rd":
+			if ds.Viz != nil {
```

Comment:
Could you add a comment on why Viz is handled differently here.

Reply:
"rd" is a stand in for "rendered", which isn't a component. I'm going to remove support for dropping `rd` for now, as it's not a component